Commands/Public/roleinfo.js

```diff
@@ -0,0 +1,93 @@
+const moment = require("moment");
+const PaginatedEmbed = require("../../Modules/MessageUtils/PaginatedEmbed");
+const Permissions = require("discord.js/src/util/Permissions.js");
+
+const rolesPerPage = 25;
+
+module.exports = async ({ client, Constants: { Colors, Text }, Utils: { TitlecasePermissions } }, documents, msg, commandData) => {
+	if (!msg.suffix) {
+		const guildRoles = [...msg.guild.roles.values()].sort((a, b) => b.position - a.position);
+		const descriptions = [];
+		for (let i = 0; i < guildRoles.length; i += rolesPerPage) {
+			const roleSegment = guildRoles.slice(i, i + rolesPerPage).join("\n");
+			descriptions.push(`These are the roles on this server:\n\n${i ? `...${i} previous roles\n` : ""}${roleSegment}${i + rolesPerPage < guildRoles.length ? `\n...and ${guildRoles.length - i - rolesPerPage} more` : ""}`);
+		}
+
+		const memberRoles = [...msg.member.roles.values()].sort((a, b) => b.position - a.position);
+		const totalPermissionsBitfield = memberRoles.reduce((a, b) => a | b.permissions, 0); // eslint-disable-line no-bitwise
```

Comment:
What is this.. Member.permissions

Reply:
It's not... This is Role.permissions

Check out `msg.member.roles.values()`.